.github/workflows/hardware-smoke.yml

```diff
@@ -236,3 +236,207 @@ jobs:
+
+      - name: Install clang 17
+        if: ${{ steps.metal-cache-restore.outputs.cache-hit != 'true' }}
+        run: |
+          wget https://apt.llvm.org/llvm.sh
+          sed -i "s/http:\/\/apt.llvm.org/https:\/\/apt.llvm.org/" -i llvm.sh
+          chmod +x llvm.sh
+          ./llvm.sh 17 all
+          rm -f llvm.sh
+
+      - name: Clone Metal and Install Dependencies
+        shell: bash
+        run: |
+          west config manifest.group-filter -- +ci
+          west update
+          cd ci/tt-metal
+          sed -i "s/http:\/\/apt.llvm.org/https:\/\/apt.llvm.org/" -i install_dependencies.sh
+          ./install_dependencies.sh --mode build --docker
+          echo "ARCH_NAME=blackhole" >> "$GITHUB_ENV"
+          echo "TT_METAL_HOME=$PWD" >> "$GITHUB_ENV"
+          echo "PYTHONPATH=$PWD" >> "$GITHUB_ENV"
+          echo "LD_LIBRARY_PATH=$PWD/build_Release/lib" >> "$GITHUB_ENV"
+
+      - name: Restore Metal Build Cache
+        id: metal-cache-restore
+        uses: actions/cache/restore@v4
+        with:
+          path: |
+            ci/tt-metal/build_Release/programming_examples
+            ci/tt-metal/build_Release/test
+            ci/tt-metal/build_Release/lib
+            ci/tt-metal/python_env
+            ci/tt-metal/runtime
+          key: metal-build-cache-${{ hashFiles('tt-zephyr-platforms/submanifests/ci.yaml') }}
+
+      - name: Generate board names
+        shell: bash
+        run: |
+          case "${{ matrix.config.board }}" in
+            p100) SMC_BOARD=tt_blackhole@p100/tt_blackhole/smc;;
+            p100a) SMC_BOARD=tt_blackhole@p100a/tt_blackhole/smc;;
+            p150a) SMC_BOARD=tt_blackhole@p150a/tt_blackhole/smc;;
+            p300a) SMC_BOARD=tt_blackhole@p300a/tt_blackhole/smc;;
+            *) echo "Unknown board: ${{ matrix.config.board }}"; exit 1;;
+          esac
+          case "${{ matrix.config.board }}" in
+            p100) BMC_BOARD=tt_blackhole@p100/tt_blackhole/bmc;;
+            p100a) BMC_BOARD=tt_blackhole@p100a/tt_blackhole/bmc;;
+            p150) BMC_BOARD=tt_blackhole@p150/tt_blackhole/bmc;;
+            p300) BMC_BOARD=p300/tt_blackhole/bmc;;
+            *) echo "Unknown board: ${{ matrix.config.board }}"; exit 1;;
+          esac
+          echo "SMC_BOARD=$SMC_BOARD" >> "$GITHUB_ENV"
+          echo "BMC_BOARD=$BMC_BOARD" >> "$GITHUB_ENV"
+
+      - name: Flash Firmware
+        working-directory: tt-zephyr-platforms
+        run: |
+          # Rescan PCIe bus to be sure we can access card
+          ./scripts/rescan-pcie.sh
+          sleep 5
+          west build -p always -b ${SMC_BOARD} --sysbuild app/smc
+          west flash -r tt_flash --force
+
+      - name: Prepare Metal Tests
+        working-directory: ci/tt-metal
+        shell: bash
+        run: |
+          # Place _ttnn.so in the correct location
+          cp build_Release/lib/_ttnn.so ttnn/ttnn/_ttnn.so
+          # Symlink build dir to build_Release
+          ln -s build_Release build
```

Comment:
What about using or appending to `LD_LIBRARY_PATH` in `GITHUB_ENV`?

Reply:
I'm not sure why, but I needed to symlink the build dir as well. I suspect that some of the metal test scripts expect to find the `build` folder

Reply:
Yeah, I think that would be awesome! I'd prefer it to be built on each PR though- that way we can pull the container from PRs when there is a change (like harvesting) that needs to be tested with a new firmware version prior to both merging